internal/provider/job_action_test.go

```diff
@@ -0,0 +1,130 @@
+package provider
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+func TestAccAAPJobAction_basic(t *testing.T) {
+	jobTemplateID := os.Getenv("AAP_TEST_JOB_TEMPLATE_ID")
+	randNum, _ := rand.Int(rand.Reader, big.NewInt(50000000))
+	inventoryName := fmt.Sprintf("%s-%d", "tf-acc", randNum.Int64())
+	timestamp := time.Now().Format("20060102150405")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccJobResourcePreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBasicJobAction(inventoryName, jobTemplateID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccJobLaunched(t, timestamp),
+				),
+			},
+		},
+	})
+}
+
+func getAllJobResults(results []interface{}, url string) ([]interface{}, error) {
+	body, err := testMethodResource(http.MethodGet, url)
+	if err != nil {
+		return nil, err
+	}
+
+	var data map[string]any
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return nil, err
+	}
+	r, ok := data["results"].([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("results field not found or not an array")
+	}
+	results = append(results, r...)
+
+	if data["next"] != nil {
+		next, ok := data["next"].(string)
+		if !ok {
+			return nil, fmt.Errorf("next field not found or not a string")
+		}
+		return getAllJobResults(results, next)
+	}
+	return results, nil
+}
+
+func testAccJobLaunched(_ *testing.T, timestamp string) resource.TestCheckFunc {
+	return func(_ *terraform.State) error {
+		results, err := getAllJobResults(nil, fmt.Sprintf("api/controller/v2/jobs?unified_job_template=%s", os.Getenv("AAP_TEST_JOB_TEMPLATE_ID")))
+		if err != nil {
+			return err
+		}
+
+		targetTime, err := time.Parse("20060102150405", timestamp)
+		if err != nil {
+			return fmt.Errorf("failed to parse timestamp: %v", err)
+		}
+
+		for _, result := range results {
+			resultMap, ok := result.(map[string]interface{})
+			if !ok {
+				continue
+			}
+			createdStr, ok := resultMap["created"].(string)
+			if !ok {
+				continue
+			}
+
+			jobTemplateID, ok := resultMap["summary_fields"].(map[string]interface{})["job_template"].(map[string]interface{})["id"].(float64)
+			if !ok {
+				continue
+			}
+			if fmt.Sprintf("%v", jobTemplateID) != os.Getenv("AAP_TEST_JOB_TEMPLATE_ID") {
+				continue
+			}
+
+			// Parse the job creation time (assuming RFC3339 format from API)
+			jobTime, err := time.Parse(time.RFC3339, createdStr)
+			if err != nil {
+				continue
+			}
+
+			// Check if job was created within 5 minutes of the target timestamp
+			timeDiff := jobTime.Sub(targetTime).Abs()
```

Comment:
Id be open to thoughts on if there's a better way to determine if an action, which isn't written to state kicks off. I was able to see the action start in the ui, but I could see where maybe this produces a false positive if another job is launched with this template id.